src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1252,6 +1252,18 @@
   (jdbc/do-commands
     "create index resource_params_cache_parameters_idx on resource_params_cache using gin (parameters)"))
 
+(defn fix-missing-edges-fk-constraint []
+  (when-not (sutils/constraint-exists? "certnames" "edges_certname_fkey")
+    (log/info "Cleaning up orphaned edges")
+    (jdbc/delete! :edges ["certname not in (select certname from certnames)"])
```

Comment:
I think it might be better to grab the valid rows into a temp table, truncate edges, and then re-insert the valid rows.

If we delete 50% of the rows we'll be leaving behind some pretty bloated tables and indexes.  If we truncate and reinsert we'll be leaving behind a really full not-bloated table.